In [1]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
%matplotlib inline
df = pd.read_csv('C://duh/Thinkful/welcome/WELLCOME_APCspend2013_forThinkful.csv', engine='python')

In [2]:
# Rename column from "COST (£) charged to Wellcome (inc VAT when charged)" to "Cost" to make coding easier
df = df.rename(index=str, columns={"PMID/PMCID":"PMID/PMCID", "Publisher":"Publisher", "Journal title":"Journal title", "Article title":"Article title", "COST (£) charged to Wellcome (inc VAT when charged)":"Cost"})
# Strip whitespace from Journal Titles
df['Journal title'] = df['Journal title'].str.strip()
# Standardize spelling of Journal Titles
df['Journal title'] = df['Journal title'].str.capitalize()
# Strip whitespace from Cost
df['Cost'] = df['Cost'].str.strip()
# Remove £ from Cost
df['Cost'] = df['Cost'].str.replace('£', '')
df['Cost'] = df['Cost'].str.replace('$', '')
df['Cost'] = pd.to_numeric(df['Cost'], downcast='float')

In [3]:
# Group the top 5 journals by article count
df.groupby('Journal title').count().nlargest(5,'Article title')

PMID/PMCID  Publisher  Article title  Cost
Journal title                                                              
Plos one                                188        190            190   190
Journal of biological chemistry          52         53             53    53
Neuroimage                               28         29             29    29
Nucleic acids research                   25         26             26    26
Plos genetics                            23         24             24    24

## Most Common Journals and Article Count
1. Plos One - 190
2. Journal of Biological Chemistry - 53
3. Neuroimage - 29
4. Nucleic Acids Research - 26
5. Plos Genetics - 24

In [4]:
print('Description of Plos One costs:')
df[df['Journal title'] == 'Plos one']['Cost'].describe()

Description of Plos One costs:


count       190.000000
mean      38751.285156
std      189009.515625
min         122.309998
25%         852.455002
50%         898.475006
75%        1044.737488
max      999999.000000
Name: Cost, dtype: float64

In [5]:
print('Description of Journal of Biological Chemistry costs:')
df[df['Journal title'] == 'Journal of biological chemistry']['Cost'].describe()

Description of Journal of Biological Chemistry costs:


count        53.000000
mean      20264.632812
std      137165.484375
min         265.670013
25%        1152.719971
50%        1314.530029
75%        1610.310059
max      999999.000000
Name: Cost, dtype: float64

In [6]:
print('Description of Neuroimage costs:')
df[df['Journal title'] == 'Neuroimage']['Cost'].describe()

Description of Neuroimage costs:


count      29.000000
mean     2215.168457
std       266.653961
min      1747.160034
25%      2100.540039
50%      2326.429932
75%      2396.570068
max      2503.340088
Name: Cost, dtype: float64

In [7]:
print('Description of Nucleic Acids Research costs:')
df[df['Journal title'] == 'Nucleic acids research']['Cost'].describe()

Description of Nucleic Acids Research costs:


count      26.00000
mean     1149.00000
std       442.94043
min       710.00000
25%       852.00000
50%       852.00000
75%      1704.00000
max      2184.00000
Name: Cost, dtype: float64

In [8]:
print('Description of Plos Genetics costs:')
df[df['Journal title'] == 'Plos genetics']['Cost'].describe()

Description of Plos Genetics costs:


count        24.000000
mean      84839.437500
std      281865.718750
min        1394.050049
25%        1484.862518
50%        1718.390015
75%        1778.224976
max      999999.000000
Name: Cost, dtype: float64

In [9]:
df['PMID/PMCID'] = df['PMID/PMCID'].str.strip()
df.groupby('PMID/PMCID').mean()

Cost
PMID/PMCID                                           
(PMCID:PMC3757918)                      999999.000000
-                                         2087.782959
19324410                                  2363.139893
20800751 PMC3778978                       2174.850098
20975956                                   850.000000
21255266 PMC3664024                       2408.350098
21483485                                  1404.000000
21624095  PMCID: PMC3734623               1991.500000
21680110 PMC3629561                       2470.860107
21903599                                  2040.000000
21948184 PMC3528370                       2338.280029
22155499 PMC3381227                       2152.760010
22241985                                  1440.000000
22302008 PMC3405838                        601.099976
22364555                                  1834.770020
22533473                                  1800.000000
22566126 PMC3746124                       2380.959961
22591621 PMC3427858                       2312.300049
22610094                                   265.670013
22618994 PMC344470                        1981.189941
22653709 PMC3781703                       2380.959961
22716184  PMC3440593                      1863.000000
22735079                                  1480.000000
22738332 PMC3381227                       1350.000000
22777780 PMCID: PMC3568905                1853.000000
22807091  PMC3744754                      1500.000000
22875964 PMCID: PMC3457148                1247.410034
22897233 PMC3599488                       2270.500000
22897899 MC3489778                        1117.369995
22901061 PMC3533787                       1982.810059
...                                               ...
PMID:24047602                             2250.000000
PMID:24048963 (Epub Sept 2013)            2841.600098
PMID:24056298                             1464.880005
PMID:24118397                             2257.560059
PMID:24125554                             2347.399902
PMID:24132152   PMC3821636                1384.339966
PMID:24150688                             2209.760010
PMID:24285889                             2346.179932
Pending                                   1071.290039
Pending - Epub April 2013                 1700.000000
Pending PMID                               886.299988
Print not yet pub?                        3000.000000
Pub July 2013                             2445.736572
Pub online first July 2013                1500.000000
Published 3 July 13                       1680.000000
Published Sept 2013                       2351.060059
e-pub                                     2329.800049
epub                                      2388.080078
in press                                  2400.000000
in press - due August 13                  2034.000000
not yet published                         1632.169922
not yet published in print                2061.820068
print due Sept 13                         1500.000000
print in press                            1921.130005
print not yet published                   1903.030029
print published August 2013               1970.569946
pub Aug 2013                              1704.203369
published August 2013                     1954.739990
published Sept 2013                       1500.000000
published jun 2013 epub ahead of print    1500.000000

[1877 rows x 1 columns]